packages/block-library/src/columns/edit.native.js

```diff
@@ -0,0 +1,279 @@
+ *
+ * @constant
+ * @type {string[]}
+ */
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+/**
+ * Number of columns to assume for template in case the user opts to skip
+ * template option selection.
+ *
+ * @type {number}
+ */
+const DEFAULT_COLUMNS = 2;
+const MIN_COLUMNS_NUMBER = 1;
+
+const BREAKPOINTS = {
+	mobile: 480,
+	large: 768,
+};
+
+function ColumnsEditContainer( {
+	attributes,
+	updateAlignment,
+	updateColumns,
+	columnCount,
+	isSelected,
+	onAddNextColumn,
+	onDelete,
+} ) {
+	const [ resizeListener, sizes ] = useResizeObserver();
+	const [ columnsSettings, setColumnsSettings ] = useState( {
+		columnsInRow: MIN_COLUMNS_NUMBER,
+	} );
+
+	const { verticalAlignment } = attributes;
+	const { width } = sizes || {};
+
+	useEffect( () => {
+		const newColumnCount = ! columnCount ? DEFAULT_COLUMNS : columnCount;
+		updateColumns( columnCount, newColumnCount );
+		setColumnsSettings( {
+			...columnsSettings,
+			columnsInRow: getColumnsInRow( width, newColumnCount ),
+		} );
+	}, [ columnCount ] );
+
+	useEffect( () => {
+		setColumnsSettings( {
+			columnsInRow: getColumnsInRow( width, columnCount ),
+			width,
+		} );
+	}, [ width ] );
+
+	const getColumnsInRow = ( containerWidth, columnsNumber ) => {
+		if ( containerWidth < BREAKPOINTS.mobile ) {
+			// show only 1 Column in row for mobile breakpoint container width
+			return 1;
+		} else if ( containerWidth < BREAKPOINTS.large ) {
+			// show 2 Column in row for large breakpoint container width
+			return Math.min( columnCount, 2 );
+		}
+		// show all Column in one row
+		return columnsNumber;
+	};
+
+	const renderAppender = () => {
+		if ( isSelected ) {
+			return (
+				<InnerBlocks.ButtonBlockAppender
+					customOnAdd={ onAddNextColumn }
+				/>
+			);
+		}
+		return null;
+	};
+
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Columns Settings' ) }>
+					<StepperControl
+						label={ __( 'Number of columns' ) }
+						icon="columns"
+						value={ columnCount }
+						onChange={ ( value ) =>
+							updateColumns( columnCount, value )
+						}
+						min={ MIN_COLUMNS_NUMBER }
+						max={ columnCount + 1 }
+						separatorType={ 'none' }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<BlockControls>
+				<BlockVerticalAlignmentToolbar
+					onChange={ updateAlignment }
+					value={ verticalAlignment }
+				/>
+			</BlockControls>
+			<View style={ isSelected && styles.innerBlocksSelected }>
+				{ resizeListener }
+				<InnerBlocks
+					renderAppender={ renderAppender }
+					__experimentalMoverDirection={
+						getColumnsInRow( width, columnCount ) > 1
+							? 'horizontal'
+							: undefined
+					}
+					flatListProps={ {
+						contentContainerStyle: styles.columnsContainer,
+						horizontal: true,
+						scrollEnabled: false,
+						style: styles.innerBlocks,
+					} }
+					allowedBlocks={ ALLOWED_BLOCKS }
+					customBlockProps={ {
+						...columnsSettings,
+						readableContentViewStyle: { flex: 1 },
```

Comment:
Could you tell me more about "readableContentViewStyle"? In what other cases we need to set this one? Is it something that can be decided implicitly rather than passing from outside?

Reply:
To allow stretch Column placed side by side I need to set `flex: 1`.  Se below screen:
<image src="https://user-images.githubusercontent.com/21242757/78263627-7fa31500-7502-11ea-9fac-7cd735221d65.png" width="500" />


>Is it something that can be decided implicitly rather than passing from outside?

You mean we should set it directly inside `readableContentView` ?

Reply:
I understand but there's a similar kind of problem here, consumer of InnerBlocks shouldn't need to know there's a readableContentView component inside. We should instead represent it with a meaningful prop, or, make this decision inside if possible. So I was wondering in what other cases it needs stretching like that? Can we say that "other than root level we need to stretch it"?

Reply:
Currently this is Column specific behaviour (in e.g. Buttons as a horizontal block **shouldn't** be stretched this way even they also are placed side by side)

> make this decision inside if possible

We can try to check if BlockListBlock that we render inside `renderItem()` in BlockList can have children (`Button` block can not have children - `Column` can ) and based on that apply `flex: 1` on `readableContentView` internally. However I'm not sure if that condition will applies to all block added in future

>We should instead represent it with a meaningful prop

How about `wrapperStyle` ?It will be more generic and do not point that we use ReadableContentView

Reply:
> Currently this is Column specific behaviour (in e.g. Buttons as a horizontal block shouldn't be stretched this way even they also are placed side by side)

Ok then, how does it sound to introduce a prop like "contentResizeMode"? It can have values like "none", "stretch" etc.? It can default to "none" but Columns can pass "stretch".

Reply:
I go with that approach please check how it looks now